api/tests/unit/permissions/permission_service/test_get_permitted_environments_for_user.py

```diff
@@ -0,0 +1,177 @@
+    VIEW_ENVIRONMENT,
+    VIEW_IDENTITIES,
+)
+from environments.permissions.models import EnvironmentPermissionModel
+from permissions.permission_service import get_permitted_environments_for_user
+
+
+def test_get_permitted_environments_for_user_returns_all_environments_for_org_admin(
+    admin_user, environment, project, project_two_environment
+):
+    for permission in EnvironmentPermissionModel.objects.all().values_list(
+        "key", flat=True
+    ):
+        # Then
+        assert (
+            get_permitted_environments_for_user(admin_user, project, permission).count()
+            == 1
+        )
+
+
+@pytest.mark.parametrize(
+    "project_admin",
+    [
+        (lazy_fixture("project_admin_via_user_permission")),
+        (lazy_fixture("project_admin_via_user_permission_group")),
+        (lazy_fixture("project_admin_via_user_role")),
+        (lazy_fixture("project_admin_via_group_role")),
+    ],
+)
+def test_get_permitted_environments_for_user_returns_all_the_environments_for_project_admin(
+    test_user, environment, project, project_admin, project_two_environment
+):
+    for permission in EnvironmentPermissionModel.objects.all().values_list(
+        "key", flat=True
+    ):
+        # Then
+        assert (
+            get_permitted_environments_for_user(test_user, project, permission).count()
+            == 1
+        )
+
+
+@pytest.mark.parametrize(
+    "environment_admin",
+    [
+        (lazy_fixture("environment_admin_via_user_permission")),
+        (lazy_fixture("environment_admin_via_user_permission_group")),
+        (lazy_fixture("environment_admin_via_user_role")),
+        (lazy_fixture("environment_admin_via_group_role")),
+    ],
+)
+def test_get_permitted_environments_for_user_returns_the_environment_for_environment_admin(
+    test_user, environment, project, environment_admin, project_two_environment
+):
+    for permission in EnvironmentPermissionModel.objects.all().values_list(
+        "key", flat=True
+    ):
+        # Then
+        assert (
+            get_permitted_environments_for_user(test_user, project, permission).count()
+            == 1
+        )
+
+
+def test_get_permitted_environments_for_user_returns_correct_environment(
+    test_user,
+    environment,
+    project_two_environment,
+    project,
+    environment_permission_using_user_permission,
+    environment_permission_using_user_permission_group,
+    environment_permission_using_user_role,
+    environment_permission_using_group_role,
+):
+    # First, let's assert that the user does not have access to any environment
+    for permission in EnvironmentPermissionModel.objects.all().values_list(
+        "key", flat=True
+    ):
+        assert (
+            get_permitted_environments_for_user(test_user, project, permission).count()
+            == 0
+        )
+    # Next, let's give user some permissions using `user_permission`
+    environment_permission_using_user_permission.permissions.add(VIEW_ENVIRONMENT)
```

Comment:
**Minor**

I think we could perhaps simplify this as follows: 

```
permissions_as_user = [VIEW_ENVIRONMENT, UPDATE_FEATURE_STATE]
environment_permission_using_user_permission.permissions.add(*permissions_as_user)
```

Then we can reuse this list below.